library/src/main/java/com/wultra/android/mtokensdk/api/operation/model/mobiletokendata/MobileTokenDataBuilder.kt

```diff
@@ -0,0 +1,78 @@
+/*
+ * Copyright 2025 Wultra s.r.o.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions
+ * and limitations under the License.
+ */
+
+package com.wultra.android.mtokensdk.api.operation.model.mobiletokendata
+
+import com.wultra.android.mtokensdk.api.operation.model.IOperation
+
+/**
+ * Helper for composing additional data passed with an [IOperation].
+ *
+ * The builder combines generic key–value pairs with structured records
+ * that contribute a single top-level entry.
+ */
+object MobileTokenData {
+
+    /**
+     * Builds a map of additional data composed from generic entries
+     * and structured [MobileTokenDataRecord] instances.
+     *
+     * Thread-safe: all mutations are synchronized.
+     * Replacement semantics: putting the same key overwrites the prior value.
+     */
+    class Builder(
+        /** Optional initial entries inserted into the builder. */
+        initialData: Map<String, Any>? = null
```

Comment:
This could be a directly empty Map, and the apply inthe  next property could be avoided.